src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -82,20 +82,31 @@ public void eventsMessage(final EventDescription eventDescription) {
 
     // Must be thread safe.
     public void peopleMessage(final PeopleDescription peopleDescription) {
-        final Message m = Message.obtain();
-        m.what = ENQUEUE_PEOPLE;
-        m.obj = peopleDescription;
+        try {
+            peopleDescription.getMessage().toString();
 
-        mWorker.runMessage(m);
+            final Message m = Message.obtain();
+            m.what = ENQUEUE_PEOPLE;
+            m.obj = peopleDescription;
+
+            mWorker.runMessage(m);
+        } catch (AssertionError e) {
+            MPLog.e(LOGTAG, "Could not record people update (see https://github.com/mixpanel/mixpanel-android/issues/567)", e);
+        }
     }
 
     // Must be thread safe.
     public void groupMessage(final GroupDescription groupDescription) {
-        final Message m = Message.obtain();
-        m.what = ENQUEUE_GROUP;
-        m.obj = groupDescription;
-
-        mWorker.runMessage(m);
+        try {
+            groupDescription.getMessage().toString();
```

Comment:
The same, check in `MixpanelMessageDescription` constructor should already handle it.